indico/modules/events/registration/fields/base.py

```diff
@@ -8,14 +8,52 @@
 from copy import deepcopy
 from decimal import Decimal
 
-from marshmallow import fields, validate
+from marshmallow import ValidationError, fields, validate, validates
 
 from indico.core.marshmallow import mm
 from indico.modules.events.registration.models.registrations import RegistrationData
 from indico.util.marshmallow import not_empty
 
 
-class BillableFieldDataSchema(mm.Schema):
+class FieldSetupSchemaBase(mm.Schema):
+    show_if_field_id = fields.Integer(required=False, load_default=None)
+    show_if_field_value = fields.String(required=False)
+
+    @validates('show_if_field_id')
+    def _check_if_field_id(self, field_id, **kwargs):
+        from indico.modules.events.registration.models.form_fields import RegistrationFormItem
+        if field_id is None:
+            return
+        field = self.context['field']
+        if field.is_manager_only or (field.parent is not None and field.parent.is_manager_only):
+            raise ValueError('Manager-only fields cannot be conditionally shown')
+        used_field_ids = set()
+        if field.id is not None:
+            if field.id == field_id:
+                raise ValueError('The field cannot conditionally depend on itself to be shown')
+            used_field_ids.add(field.id)
+        regform = self.context['regform']
+        is_same_regform = (RegistrationFormItem.query.filter_by(
+            id=field_id, registration_form_id=regform.id).scalar() is not None)
+        if not is_same_regform:
```

Comment:
Can be made more concise:

```suggestion
        if not RegistrationFormItem.query.filter_by(id=field_id, registration_form_id=regform.id).has_rows():
```

(yes, i really do not like the wrapping of the current query ;))

Reply:
I personally don't like things like:

```python
if not <insert here your very long conditional statement that likely occupies more than one line>:
```

Yet at the end of the day it's just a matter of personal preferences. I just changed it.

Reply:
I usually make the threshold at "does it fit in a single line". If it does, fine, if it doesn't, separate statement.